Automattic/wp-calypso
# Update email upsell button to link directly to email addition page

#### Proposed Changes

Currently, the **"Create a new mailbox"** button in the Embedded Inbox links out to **Upgrades > Email** `/email/:site`, where users will need to make two more clicks (select a domain, click on "Add new mailboxes") before they get to the email addition screen:

<img width="1122" alt="Screen Shot 2022-06-28 at 3 20 00 PM" src="https://user-images.githubusercontent.com/104910361/176271403-9cbaeb41-73bc-4806-b9ff-330ee30220a9.png">

The proposed change improves UX by linking users **directly to the email addition page**, provided that there is **a single domain** associated with the domain.

<img width="1102" alt="Screen Shot 2022-06-28 at 3 20 18 PM" src="https://user-images.githubusercontent.com/104910361/176268738-fc3498cc-c95c-4cb0-9040-8defcce6a10d.png">

Also took the opportunity to separate out `NewMailboxUpsell` into its own file, so things look cleaner with the additional logic for generating a different click through URL for the button. 

Ideally, I would love to have `<NewMailboxUpsell>` added via `/inbox/index.jsx` rather than adding it as part of `<MailboxSelectionList>`, since it isn't really a mailbox list item and can be placed anywhere on the page. But for now, I left upsell notice as-is, since we'd want the notice to appear at the same time as the mailbox list.

#### Testing Instructions

**Scenario A:** User has multiple domains associated with the site
1. Have multiple domains
2. Subscribe to an email service
3. Click on **"Inbox"** in the sidebar
4. Click on **"Create a new mailbox"** button

**Expected outcome:** Users land on the **Upgrades > Email** page `/email/:site`, where they need to select a domain.

---

**Scenario B:** User has a single domain associated with the site, and is subscribed to Professional Email
1. Have a single domain
2. Subscribe to Professional Email service
3. Click on **"Inbox"** in the sidebar
4. Click on **"Create a new mailbox"** button

**Expected outcome:** Users land on the email addition page for Titan `/email/:domain/titan/new/:site?source=inbox`

---

**Scenario C:** User has a single domain associated with the site, and is subscribed to Google Workspace
1. Have a single domain
2. Subscribe to Google Workspace service
3. Click on **"Inbox"** in the sidebar
4. Click on **"Create a new mailbox"** button

**Expected outcome:** Users land on the email addition page for Google `/email/:domain/google-workspace/add-users/:site?source=inbox`
